### PR TITLE
Added dependency resolver

### LINKED ISSUE #7: Dependency Resolution
As AUR packages may have dependencies on other AUR packages we should make sure the dependencies are built first.

The built dependencies then also need to be made available to the builder/executor this can be done by adding the AAAAA-generated repo to the builders and/or having a shared pacman cache

### PR DESCRIPTION
**Pull request by timanema**
Closes #7

### CODE REVIEW

**Comment by timanema** [comment]
Not quite done, as I still need to add some extra tests and documentation (and need to re-add run files), but the main code is there. I think you should be able to already take a look at it @NULLx76 @jonay2000

File: .run/Aur package cache.run.xml [hunk 1]
```diff
@@ -1,12 +0,0 @@
-<component name="ProjectRunConfigurationManager">
```
Comment (jdonszelmann) [inline .run/Aur package cache.run.xml:1]: Why did you delete these run configurations? I think they're quite useful to include

File: pkg/aur/cache.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+func SendCachedInfoRequest(url, pkg string) (res InfoResult, err error) {
+	resp, err := http.Get(fmt.Sprintf(url, pkg))
+	if err != nil {
+		return res, errors.Wrap(err, "received error from aur cache")
+	}
+	if resp.StatusCode == http.StatusNoContent {
+		return InfoResult{}, NotInAurErr
+	}
+	if resp.StatusCode != http.StatusOK {
+		return InfoResult{}, errors.New(fmt.Sprintf("received error from aur cache: %s", resp.Status))
```
Comment (jdonszelmann) [inline pkg/aur/cache.go]: fmt.Errorf

**Review by jdonszelmann — APPROVED** [review]

File: .run/Serve frontend.run.xml [hunk 3]
```diff
@@ -1,13 +0,0 @@
-<component name="ProjectRunConfigurationManager">
```
Comment (NULLx76) [inline .run/Serve frontend.run.xml:1]: please add these back

File: pkg/aur/cache.go (hunk 2 above)
Comment (NULLx76) [reply pkg/aur/cache.go]: errors.Errorf

File: pkg/aur/cache.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	"net/http"
+)
+
+// NotInAurErr will be returned when the package is not found in the AUR, meaning this dependency
+// should not be considered for resolving dependencies.
+var NotInAurErr = errors.New("pkg not in aur")
+
+// SendCachedInfoRequest will do a HTTP GET request to the given url, after replacing the '%s' with the given package
+// name. It expects to receive a InfoResult, or a 204 No Content (to signal that the package is not on the AUR).
+func SendCachedInfoRequest(url, pkg string) (res InfoResult, err error) {
```
Comment (NULLx76) [inline pkg/aur/cache.go]: The named returns are barely used?

File: services/aur_cache/main.go [hunk 5]
```diff
@@ -65,6 +65,11 @@ func info(cache store.Cache) http.HandlerFunc {
 
 		entry, err := cache.GetInfoEntry(name)
 		if err == nil {
+			if !entry.OnAur {
+				http.Error(w, "no results", http.StatusNoContent)
```
Comment (NULLx76) [inline services/aur_cache/main.go]: StatusNoContent isn't really an error (and shouldn't have content)

Comment (timanema) [reply services/aur_cache/main.go]: How about a 404? That one would make sense to me

**Review by jdonszelmann — APPROVED** [review]

**Review by NULLx76 — APPROVED** [review]
